### ISSUE TITLE
Presets for trade agreement screen

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#3509]
depending on how many there are, we might want to hard-code them or do an import/export interface similar to `stockpiles`

ideas for presets:
- military-grade bars
- military-grade ores
- all economic stones

**Comment by DragonEyeNinja** [comment]
- cheapest leather
- all leather
- cheapest sheets
- all sheets
- book bindings & scroll rollers
- all logs
- all magma-safe stones & blocks
- valuable bars or valuable alloy-possible bars & ores
- seeds (underground crops)
- seeds (aboveground crops)
- alcohols
- food & ingredients for food
- animals of note (e.g. milkable, shearable, powerful war animals like elephants)
- writing material (scrolls & codices)
- shell-giving fish
- lye, potash, pearlash, fuel
- instruments
- raw glass & sand
- dye
- thread & cloth
- foreign weapons

**Comment by myk002** [comment]
all good ideas, though shell-giving fish probably won't be useful for its intended purpose. shellfish that come from caravans is already prepared, and you don't get a shell from them.

**Comment by DragonEyeNinja** [comment]
may also want to consider allowing users to create their own new presets; ideally these should just come with the default DFHack package

**Comment by myk002** [comment]
yeah, that would be the `stockpiles` library model, which is totally viable. we'd just need to write another protobuf (or json) serialization/deserialization routine

**Comment by myk002** [comment]
other bits I don't want to forget on the trade screens:
- fix value calculation for cages with units inside
- option for listing items within bins (and other containers?) in the bring to depot screen

**Comment by myk002** [comment]
notes from [discord](https://discord.com/channels/793331351645323264/807444467140788254/1144090642662817843):

Container Mode:
You see each container, either inside the container or not. You can only select the entire container at once, and the "marked" icon is on the container line, no change if collapsed or not. It will bring the entire container.

If group is enabled, expanded container will show groups WITHIN the container. If ungrouped, each item in the container is shown separately.

Item Mode:
You see items, ignoring containers. nothing to collapse or expand. If you select items, those items are marked for delivery, leaving the bins.

Grouping works as present.
------------------------- 
^^This^^ seems to hit most use cases I can think of. Either people want to bring the items or the containers. If people just want the items they likely don't care what container it is in, so don't show/bring the container. Just the items. If people want the containers, show them container mode and only allow marking containers.
Friendly Mushrooom — Today at 7:15 PM
If I wanted every gem and every bin with gloves:
Container mode>"Gloves">Mark all
Switch to Item mode, sends request and resets the index
Item Mode>"Gem">Mark all

**Comment by myk002** [comment]
also ensure "trade goods" preset is achievable: https://gist.github.com/ab9rf/4740564
